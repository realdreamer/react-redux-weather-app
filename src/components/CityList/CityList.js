import React from 'react';
import { Link } from 'react-router-dom';

import './styles/CityList.css';

const CityList = ( {city} ) => {
  if (city.length > 0 ) {
    return (<div className="city-list">
            { city.map((city) => {
                return (
                  <div className="city-name" key={city.woeid}>
                    <Link to={"/location/"+city.woeid} className="city-link">{city.title}</Link>
                  </div>
                )
              })
            }
          </div>
    );
  }
  else {
    return <div className="no-results">No city found</div>;
  }
};

export default CityList;
