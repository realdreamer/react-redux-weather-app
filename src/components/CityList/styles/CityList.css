.no-results {
  background-color: tomato;
  color: #fff;
  display: block;
  font-size: 16px;
  padding: 16px;
  text-align: center;
  width: 100%; }

.city-link {
  background-color: #76a097;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.3);
  color: #2a4a43;
  display: flex;
  letter-spacing: 1px;
  margin-bottom: 8px;
  padding: 20px;
  text-decoration: none;
  text-transform: uppercase;
  transition: all .3s ease-in-out; }
  .city-link:hover {
    background-color: #9fcec4;
    letter-spacing: 1.6px; }
  .city-link:active {
    transform: translateY(2px); }
