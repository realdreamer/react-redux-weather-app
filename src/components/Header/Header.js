import React from "react";
import SearchForm from "../SearchForm/SearchForm";

const Header = props => (
  <header>
    <h1>Weather App</h1>
    <SearchForm onSearch={props.onSearch} />
  </header>
);

export default Header;
