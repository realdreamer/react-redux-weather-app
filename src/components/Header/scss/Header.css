.header {
  background-color: #f7b18e;
  color: #fff;
  padding: 10px;
  text-align: center; }

.header-title {
  font-size: 18px;
  text-transform: uppercase; }
