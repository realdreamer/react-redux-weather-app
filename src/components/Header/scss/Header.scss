$header-bg-color: #f7b18e;
.header {
  background-color: $header-bg-color;
  color: #fff;
  padding: 10px;
  text-align: center;
}
.header-title {
  font-size: 18px;
  text-transform: uppercase;
}
