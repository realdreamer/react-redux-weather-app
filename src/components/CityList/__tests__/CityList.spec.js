import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Header from '../CityList';
import CityList from '../CityList';
import { MemoryRouter, Link } from 'react-router-dom';

describe('<CityList /> Component ', () => {
  it('component renders without error', () => {
    const props = [];
    const tree = renderer.create(<MemoryRouter><CityList city={props}/></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('pass props with data', () => {
    const props = [{ latt_long: "13.05939 80.245667", location_type:"City", title:"Chennai", woeid:2295424 }];
    const tree = renderer.create(<MemoryRouter><CityList city={props} /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
