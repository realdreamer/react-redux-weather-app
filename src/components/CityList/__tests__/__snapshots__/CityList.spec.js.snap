// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityList /> Component  component renders without error 1`] = `
<div
  className="no-results"
>
  No city found
</div>
`;

exports[`<CityList /> Component  pass props with data 1`] = `
<div
  className="list"
>
  <div
    className="city"
  >
    <a
      href="/location/2295424"
      onClick={[Function]}
    >
      Chennai
    </a>
  </div>
</div>
`;
